File: docs/snippets/Datamaps/datamap/xlsx-file-roundtrip.ts
// docs:begin
import { DataContext, Datamap, XlsxController } from "@nfdi4plants/arctrl";

const leafArea = new DataContext(
  undefined,
  "assays/Measurement/dataset/results.csv",
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  "leaf area",
);

const datamap = new Datamap([leafArea]);

const xlsxPath =
  "docs/generated/snippets/Datamaps/datamap/datamap-file-roundtrip.xlsx";

await XlsxController.Datamap.toXlsxFileAsync(xlsxPath, datamap);

const datamapAgain = await XlsxController.Datamap.fromXlsxFileAsync(xlsxPath);
// docs:end

// docs:assert
if (datamapAgain.DataContexts.length !== 1) {
  throw new Error(
    `Expected one data context after xlsx file roundtrip, got ${datamapAgain.DataContexts.length}`,
  );
}
// docs:endassert

File: docs/snippets/Datamaps/datamap/xlsx-file-roundtrip.py
# docs:begin
from pathlib import Path

from arctrl import DataContext, Datamap, XlsxController

leaf_area = DataContext(
    name="assays/Measurement/dataset/results.csv",
    label="leaf area",
)

datamap = Datamap([leaf_area])

xlsx_path = Path(__file__).with_name("datamap-file-roundtrip.xlsx")

XlsxController.Datamap().to_xlsx_file(str(xlsx_path), datamap)

datamap_again = XlsxController.Datamap().from_xlsx_file(str(xlsx_path))
# docs:end

# docs:assert
if len(datamap_again.DataContexts) != 1:
    raise Exception(
        "Expected one data context after xlsx file roundtrip, "
        f"got {len(datamap_again.DataContexts)}"
    )
# docs:endassert
